calliope/backend/parsing.py

```diff
@@ -617,7 +620,7 @@ def evaluate_foreach(self, model_data: xr.Dataset) -> xr.DataArray:
         # Add other dimensions (costs, timesteps, etc.)
         add_dims = set(self.sets).difference(initial_imask.dims)
         if add_dims.difference(model_data.dims):
-            exceptions.warn(
```

Comment:
There is a test asserting that a warning is raised here. Does that show that we should rather keep it as a warning or shall we drop the test, @brynpickering?

Reply:
The test can just check for the logged debug-level warning